test/param/ferguson/fielderator-user-class.chpl

```diff
@@ -0,0 +1,34 @@
+use Reflection;
+
+class R {
+  var x: int;
+  var y: int;
+}
+
+
+assert(numFields(R) == 2);
+assert(getFieldName(R, 1) == "x");
+
+assert(getFieldIndex(R, "x") == 1);
+assert(getFieldIndex(R, "y") == 2);
+assert(getFieldIndex(R, "z") == 0); // no field z.
+assert(hasField(R, "x") == true);
+assert(hasField(R, "y") == true);
+assert(hasField(R, "z") == false);
+
+
+var r = new R(10,20);
+
+assert(getField(r, 1) == 10);
+assert(getField(r, 2) == 20);
+
+assert(getField(r, "x") == 10);
+assert(getField(r, "y") == 20);
+
+
+getFieldRef(r, 1) = 100;
+getFieldRef(r, "y") = 200;
+
+assert(getFieldRef(r, "x") == 100);
+assert(getFieldRef(r, "y") == 200);
+
```

Comment:
This might seem silly, but how about adding a writeln() of r here at the end, just to demonstrate that the record really does hold 100, 200 now (for the purposes of computation) and that all the manipulations haven't been in some abstract reflection-oriented world?  (This also has the nice side-effect of making the .good file non-empty which I tend to like for the sake of not being surprised... but that's truly a secondary reason for asking).